### PR TITLE
Acl xduser

### PR DESCRIPTION
**Pull request by ryanrath**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: classes/XDUser.php [hunk 1]
```diff
@@ -1,466 +1,486 @@
...
+     * function will return null.
+     * cl
+     * @return null|XDUser
+     */
+    public static function getPublicUser()
+    {
+        if (null !== self::$_publicUser) {
+            return self::$_publicUser;
+        } else {
+            self::$_publicUser = self::getUserByUserName('Public User');
```
Comment (jpwhite4) [inline classes/XDUser.php]: Where is the code that automatically creates the 'Public User' row in the database? I can't find it anywhere.

Comment (jpwhite4) [inline classes/XDUser.php]: Also as a coding standard comment: 

I prefer to not repeat identical lines in the code unless it is unavoidable. I suggest the following, which I think is easier to see what is happening:
```
if (null === self::$_publicUser) {
    self::$_publicUser = self::getUserByUserName('Public User');
}
return self::$_publicUser;
```

File: shippable.yml [hunk 2]
```diff
@@ -18,6 +18,9 @@ build:
         - composer install --no-progress
         - cp ~/assets/secrets open_xdmod/modules/xdmod/integration_tests/.secrets
         - ./open_xdmod/modules/xdmod/integration_tests/runtests.sh --log-junit `pwd`/shippable/testresults/results.xml
+        - cp ./configuration/portal_settings.ini ./configuration/portal_settings.ini.old
+        - cp -f /etc/xdmod/portal_settings.ini ./configuration/portal_settings.ini
+        - ./open_xdmod/modules/xdmod/component_tests/runtests.sh --log-junit `pwd`/shippable/testresults/componentresults.xml
+
```
Comment (jpwhite4) [inline shippable.yml]: You should restore the portal_settings.ini.old to its original location after running the tests (as we discussed before).

Comment (ryanrath) [inline shippable.yml]: Thanks! I did totally miss that. Latest commit mv's the portal_settings.ini.old back to portal_settings.ini.

**Pull request merged**